R/join-rel.R

```diff
@@ -0,0 +1,266 @@
+
+#' Append a Substrait Join relation
+#'
+#' @param compiler_left,compiler_right A [substrait_compiler()] or table-like
+#'   object to use as inputs to the join relation. Currently only one of
+#'   these can be a [substrait_compiler()].
+#' @param by A join specifiation. Join specifications that are currently
+#'   supported include a character vector of column names whose optional
+#'   names indicate the name on `compiler_left` (e.g., `"common_name"` or
+#'   `c("name_left" = "name_right")`).
+#' @param type One of JOIN_TYPE_INNER, JOIN_TYPE_OUTER, JOIN_TYPE_LEFT,
+#'   JOIN_TYPE_RIGHT, JOIN_TYPE_SEMI, JOINT_TYPE_ANTI, or JOIN_TYPE_SINGLE.
+#' @param name_repair A function of `output_mapping`, `names_left`, and
+#'   `names_right` used to calculate the output names (e.g.,
+#'   [join_name_repair_suffix_common()] or [join_name_repair_none()]).
+#' @param output_mapping A function of `by`, `names_left`, and `names_right`
+#'   used to calculate the zero-based indices of the output to include (e.g.,
+#'   [join_emit_all()] or [join_emit_default()]).
```

Comment:
Can these be rephrased from "a function of"? This is tricky to quickly mentally parse. Also, please could you add an example of using these parameters to the examples section?